src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneGetFileMethodSender.php

```diff
@@ -0,0 +1,135 @@
+use SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PayoneGetFileMethodSender implements PayoneGetFileMethodSenderInterface
+{
+    public const ERROR_ACCESS_DENIED_MESSAGE = 'Access denied';
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface
+     */
+    protected $executionAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @var \Generated\Shared\Transfer\PayoneStandardParameterTransfer
+     */
+    protected $standardParameter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Key\HmacGeneratorInterface|\SprykerEco\Zed\Payone\Business\Key\HashGenerator
+     */
+    protected $hashGenerator;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMethodMapperInterface[]
+     */
+    protected $registeredMethodMappers;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface
+     */
+    protected $standartParameterMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader
+     */
+    protected $paymentMapperReader;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface $executionAdapter
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PayoneStandardParameterTransfer $standardParameter,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PaymentMapperReader $paymentMapperReader
+    ) {
+        $this->executionAdapter = $executionAdapter;
+        $this->queryContainer = $queryContainer;
+        $this->standardParameter = $standardParameter;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->paymentMapperReader = $paymentMapperReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneGetFileTransfer $getFileTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayoneGetFileTransfer
+     */
+    public function getFile(PayoneGetFileTransfer $getFileTransfer): PayoneGetFileTransfer
+    {
+        $responseContainer = new GetFileResponseContainer();
+        $paymentEntity = $this->findPaymentByFileReferenceAndCustomerId(
+            $getFileTransfer->getReference(),
+            $getFileTransfer->getCustomerId()
+        );
+
+        if ($paymentEntity) {
```

Comment:
* Extract to a method `fetchFile(PayoneGetFileTransfer):GetFileResponseContainer`.
* Get rid from `else` condition.

Reply:
Need discussion

Reply:
++